include/parallel/parallel.h

```diff
@@ -1098,6 +1098,14 @@ class Communicator
                      T send,
                      std::vector<T> & recv) const;
 
+   /**
+    * Gather overload for string types
+    */
+   template <typename T>
+   inline void gather(const unsigned int root_id,
+                      std::basic_string<T> send,
```

Comment:
We already use const_cast to handle insanity like MPI1 MPI_Send taking a non-const void*.  If there's other took-them-a-while-to-notice-not-Fortran bits in the standard then we can use the same workaround again.  I'd say just check the MPI3 standard first; if things still aren't const-correct there then there might be an actual reason.